receiver/googlecloudpubsubreceiver/internal/log_entry.go

```diff
@@ -0,0 +1,574 @@
+		case "timestamp":
+			// timestamp -> Timestamp
+			var t time.Time
+			err = json.Unmarshal(v, &t)
+			if err != nil {
+				return res, lr, err
+			}
+			lr.SetTimestamp(pcommon.NewTimestampFromTime(t))
+			delete(src, k)
+		case "resource":
+			// resource -> Resource
+			// mapping type -> gcp.resource_type
+			// labels -> gcp.<label>
+			var protoRes monitoredres.MonitoredResource
+			err = protojson.Unmarshal(v, &protoRes)
+
+			resAttrs.EnsureCapacity(len(protoRes.GetLabels()) + 1)
+			resAttrs.PutStr("gcp.resource_type", protoRes.GetType())
+			for k, v := range protoRes.GetLabels() {
+				resAttrs.PutStr(strcase.ToSnakeWithIgnore(fmt.Sprintf("gcp.%v", k), "."), v)
+			}
+			delete(src, k)
+		case "logName":
+			var logName string
+			err = json.Unmarshal(v, &logName)
+			if err != nil {
+				return res, lr, err
+			}
+			// log_name -> Attributes[“gcp.log_name”]
+			attrs.PutStr("gcp.log_name", logName)
+			delete(src, k)
+		case "jsonPayload", "textPayload":
+			// {json,proto,text}_payload -> Body
+			var payload any
+			err = json.Unmarshal(v, &payload)
+			if err != nil {
+				return res, lr, err
+			}
+			// Note: json.Unmarshal will turn a bare string into a
+			// go string, so this call will correctly set the body
+			// to a string Value.
+			lr.Body().FromRaw(payload)
+			delete(src, k)
+		case "protoPayload":
+			// {json,proto,text}_payload -> Body
+			err = translateInto(lr.Body().SetEmptyMap(), (&anypb.Any{}).ProtoReflect().Descriptor(), v)
+			if err != nil {
+				return res, lr, err
+			}
+			delete(src, k)
+		case "severity":
+			var severity string
+			err = json.Unmarshal(v, &severity)
+			if err != nil {
+				return res, lr, err
+			}
+			// severity -> Severity
+			// According to the spec, this is the original string representation of
+			// the severity as it is known at the source:
+			//   https://opentelemetry.io/docs/reference/specification/logs/data-model/#field-severitytext
+			lr.SetSeverityText(severity)
+			delete(src, k)
+		case "trace":
+			var trace string
+			err = json.Unmarshal(v, &trace)
+			if err != nil {
+				return res, lr, err
+			}
+			lr.SetTraceID(cloudLoggingTraceToTraceIDBytes(trace))
+			delete(src, k)
+		case "spanId":
+			var spanId string
+			err = json.Unmarshal(v, &spanId)
+			if err != nil {
+				return res, lr, err
+			}
+			lr.SetSpanID(spanIDStrToSpanIDBytes(spanId))
+			delete(src, k)
+		case "labels":
+			var labels map[string]string
+			err = json.Unmarshal(v, &labels)
+			if err != nil {
+				return res, lr, err
+			}
+			// labels -> Attributes
+			for k, v := range labels {
+				attrs.PutStr(k, v)
+			}
+			delete(src, k)
+		case "httpRequest":
+			httpRequestAttrs := attrs.PutEmptyMap("gcp.http_request")
+			err = translateInto(httpRequestAttrs, getLogEntryDescriptor().Fields().ByJSONName(k).Message(), v, snakeifyKeys)
+			if err != nil {
+				return res, lr, err
+			}
+			delete(src, k)
+		default:
```

Comment:
`LogEntry.insertId` can be mapped to Attribute(`log.record.uid`) (see https://opentelemetry.io/docs/specs/otel/logs/semantic_conventions/general/)